### PR TITLE
draft svg rendering

### PR DESCRIPTION
**Pull request by gfontorbe**
Documentation about SVG rendering

### CODE REVIEW

**Review by montymxb — COMMENTED** [review]
Cool, this looks nice. Feedback is mostly some small grammar stuff.

File: hugo/content/docs/svg-rendering.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                  class-node-package={node.type === 'node:package'}
+                  class-node-class={node.type === 'node:class'}
+                  class-mouseover={node.hoverFeedback} class-selected={node.selected}
+                  x="0" y="0" width={Math.max(node.size.width, 0)} height={Math.max(node.size.height, 0)}></rect>
+            {context.renderChildren(node)}
+        </g>;
+    }
+}
+```
+The class `NodeView` extends `RectangularNodeView` which is a default `View` in Sprotty, ultimately implementing `IView`. Please note the class decorator `@injectable()`, which is necessary for the [Dependency Injection](link-to-dependency-injection) mechanism.
```
Comment (planger) [inline hugo/content/docs/svg-rendering.md:34]:
```suggestion
The class `NodeView` extends `RectangularNodeView` which is a default `View` in Sprotty, ultimately implementing `IView`. Don't forget to add the class decorator `@injectable()`, which is necessary for the [Dependency Injection](link-to-dependency-injection) mechanism.
```

File: hugo/content/docs/svg-rendering.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                  class-mouseover={node.hoverFeedback} class-selected={node.selected}
+                  x="0" y="0" width={Math.max(node.size.width, 0)} height={Math.max(node.size.height, 0)}></rect>
+            {context.renderChildren(node)}
+        </g>;
+    }
+}
+```
+The class `NodeView` extends `RectangularNodeView` which is a default `View` in Sprotty, ultimately implementing `IView`. Please note the class decorator `@injectable()`, which is necessary for the [Dependency Injection](link-to-dependency-injection) mechanism. 
+
+The `render()` method is the core of the `View`. It takes `node` as an argument of a given type, a `RenderingContext`, and an optional `args` object. It first checks if the `node` should be rendered. This is an optimization step, as we only want to render SVG elements that are inside of the viewport or are not hidden by some user-defined filter.
```
Comment (planger) [inline hugo/content/docs/svg-rendering.md:36]:
```suggestion
The `render()` method is the core of the `View`. It takes `node` -- that is the `SModelElement` to be rendered - as an argument, a `RenderingContext`, and an optional `args` object. View implementations should first check whether the `node` should be rendered at all. This is an optimization step, as we only want to render SVG elements that are inside of the visible viewport and not hidden by some other user-defined filter.
```

File: hugo/content/docs/svg-rendering.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+The `render()` method returns a `VNode` which is [Snabbdom's](https://github.com/snabbdom/snabbdom) virtual representation of a DOM element. This `VNode` can hold one and only one *root element*, therefore we need to group our SVG elements inside of a *container element* `g`.
```
Comment (planger) [inline hugo/content/docs/svg-rendering.md:37]:
```suggestion
Eventually, the `render()` method returns a `VNode` which is [Snabbdom's](https://github.com/snabbdom/snabbdom) virtual representation of a DOM element. This `VNode` can hold one and only one *root element*, therefore we need to group our SVG elements inside of a *container element* `g`.
```

**Review by planger — COMMENTED** [review]
Excellent, thank you! Some suggestions inline.

**Review by dhuebner — APPROVED** [review]
Really nice! 
I think we could add a conclusion or a "what's next" section in addition

**Pull request merged**